StyleGuide.md

```diff
@@ -970,6 +975,111 @@ Use the following HTML and Liquid to include an image in a Markdown page:
+
+- Have text be a link on the [Known Limitations](https://www.cockroachlabs.com/docs/stable/known-limitations) page.
+- Have that same text _not_ be a link on the [Cost-Based Optimizer](https://www.cockroachlabs.com/docs/stable/cost-based-optimizer) page (since it would be a self-referring link).
+
+The way to do this in Jekyll (with its templating language Liquid) is to add the following content to an include file that is shared into both pages:
+
+    {% if page.name == "cost-based-optimizer.md" %} Locality-optimized search {% else %} [Locality-optimized search](cost-based-optimizer.html#locality-optimized-search-in-multi-region-clusters) {% endif %} only works for queries selecting a limited number of records (up to 10,000 unique keys). Also, It does not yet work with [`LIMIT`](limit-offset.html) clauses.
+
+The syntax is a little hard to read inline, but based on some experimenting it appears that it _must_ be written as one line so as not to introduce line breaks in the resulting text.
+
+Formatted for easier reading, the template code looks like:
+
+```
+{% if page.name == "cost-based-optimizer.md" %}  
+Locality-optimized search  
+{% else %}  
+[Locality-optimized search](cost-based-optimizer.html#locality-optimized-search-in-multi-region-clusters)  
+{% endif %}  
+```
+
+##### Different content depending on Cockroach Cloud or Self-Hosted
+
+Sometimes, you must use an include file on both a Cockroach Cloud and a Self-Hosted page. This requires that the links be resolved differently depending on the page that the include file is being referenced from.
+
+For instructions showing how to use relative links that differ depending on the page type, see [Use relative links to Cloud and Core topics in _include files](https://cockroachlabs.atlassian.net/wiki/x/woBZl).
+
+_Note_: In the previous link, Cloud = Dedicated and Core = Self-Hosted.
+
+<a name="remote-includes"></a>
+
+##### Remote includes
+
+Sometimes you need to include files that are maintained in other places than the `cockroachdb/docs` repo but referenced in our docs. The `remote_include` tag is used for this. We most often use this tag for:
+
+- SQL diagrams, which are maintained in the `cockroachdb/cockroach` repo
+- Code samples, which are maintained in various repos
+
+For SQL diagrams, you remotely include the entire (HTML) file as follows:
+
+```
+{% remote_include https://raw.githubusercontent.com/cockroachdb/generated-diagrams/release-22.1/grammar_svg/show_databases.html %}
+```
+
+For code samples, you usually want to show only part of a larger file to highlight a specific technique, or due to length considerations.
+
+To accomplish this, the `remote_include` tag lets you pass arguements (usually named `START {text}` and `END {text}` by convention) that pull in the text of the remote file between `START {text}` and `END {text}`.
```

Comment:
```suggestion
To accomplish this, the `remote_include` tag lets you pass arguments (usually named `START {text}` and `END {text}` by convention) that pull in the text of the remote file between `START {text}` and `END {text}`.
```

Reply:
Fixed , thanks!  Forgot to run spellcheck

Also rebased onto master branch